Xamarin.Forms.Platform.Android/ResourceManager.cs

```diff
@@ -138,17 +156,48 @@ internal static Task ApplyDrawableAsync(this IShellContext shellContext, Bindabl
 			// make sure things are good before we start
 			var element = bindable ?? renderer.Element;
 
-			if (element == null || renderer.View == null)
+			if (element == null)
 				return;
 
 			onLoading?.Invoke(true);
 			if (element.GetValue(imageSourceProperty) is ImageSource initialSource && !initialSource.IsEmpty)
 			{
 				try
 				{
-					using (var drawable = await context.GetFormsDrawableAsync(initialSource, cancellationToken))
+					string cacheKey = String.Empty;
+
+					// Todo improve for other sources
+					// volley the requests better up front so that if the same request comes in it isn't requeued
+					if (initialSource is UriImageSource uri)
 					{
-						// TODO: it might be good to make sure the renderer has not been disposed
+						cacheKey = Device.PlatformServices.GetMD5Hash(uri.Uri.ToString());
+						var cache = GetCache();
+						var cacheObject = await cache.GetAsync(cacheKey);
```

Comment:
Should the use of the new cache be behind a feature flag? `LruCache_Experimental` or something?

I don't know your team's stance on when a feature flag is needed for a certain change.

Reply:
If we aren't comfortable with the LruCache going into 4.0 (next weekish) then we'd most likely just retarget the LRU part of this for master so that it can percolate inside some 4.1 pre releases first and get testing.  The main issue I was trying to combat with the LruCache is that we have a disk cache for UriSources but if multiple things request the same image from the same Url it will generate multiple bitmaps over the same stream which was causing OOM exceptions. The LruCache allows for the same underlying bitmap so it greatly streamlined cases where a user might have the same url loaded 4 or 5 times.  There's also cases where the same ImageSource might be retrigger to re-request the same image which was causing a cancel then a re-request of the same image (I realize the ImageSource here could be made more resilient) but an LruCache seemed to just be a good way to Volley all requests across different ImageSources or from the same one